### PR TITLE
Changed parser mode to 'codemod'

### LINKED ISSUE #120: Incorrect indentation on line starting with &nbsp;
Linting a template with a segment like the following incorrectly flags the indentation:

``` handlebars
{{#if foo}}
  &nbsp;bar
{{/if}}
```

with

```
block-indentation: Incorrect indentation for `   bar
` beginning at L2:C0. Expected `   bar
` to be at an indentation of 2 but was found at 3. (thedyrt-ui/templates/components/at-trip):
`{{#if foo}}
  &nbsp;bar
{{/if}}`
```

### PR DESCRIPTION
**Pull request by initram**
This allows the linter to lint indentation correctly even if the line starts with `&nbsp;`

Fixes #120

Though some changes were needed to rules that work on TextNodes as they now behave slightly different. The changes I made to the block indentation rule also made it a bit more consistent in my opinion.

Additional test may be needed to cover what ever corners cases there may be, but all current tests should pass.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -123,6 +123,7 @@ class Linter {
 
     try {
       precompile(stripBom(options.source), {
+        mode: 'codemod',
```
Comment (rwjblue) [inline lib/index.js]: 👍I absolutely had this package in mind when adding this feature upstream.

**Comment by rwjblue** [comment]
> We could consider adding more items the the default white list and test that it works for string that are currently being transformed into whitespace.

Right, I think the main thing we need to ensure is that the items in the current default whitelist that _could_ have been converted from an HTML entity have a corresponding whitelist entry for the entity as well.

**Comment by rwjblue** [comment]
@initram - Would you mind rebasing? There have been changes to `lib/index.js` which are causing conflicts, but also the definition of `codemod` mode changed in `@glimmer/syntax@0.43.0` (see https://github.com/glimmerjs/glimmer-vm/pull/983 for details on that breakage WRT `codemod` mode). We may need more tweaks to accommodate those changes.

File: lib/rules/lint-no-bare-strings.js [hunk 2]
```diff
@@ -38,8 +38,47 @@ const TAG_ATTRIBUTES = {
   img: ['alt'],
 };
 
+// Charrecter entity refeence chart: https://dev.w3.org/html5/html-author/charref
```
Comment (rwjblue) [inline lib/rules/lint-no-bare-strings.js]:
```suggestion
// Character entity reference chart: https://dev.w3.org/html5/html-author/charref
```

**Comment by rwjblue** [comment]
I went ahead and rebased for you, looks like no additional failures crop up so this is good to land!

**Comment by rwjblue** [comment]
Thank you again @initram!
